sdk/src/hooks/useOrder.ts

```diff
@@ -139,9 +156,17 @@ export function useOrder<abis extends OptionalAbis>(
     txMutation,
     waitForTx: wait,
   }
+  return status === 'initializing' || contractsResult.isError
+    ? {
+        ...orderReturn,
+        status: status === 'initializing' ? 'initializing' : 'error',
+        isReady: false,
+      }
```

Comment:
We'll need to update docs in README with new behaviour, require that users check `isReady`

Other option is to be a little more forgiving in interface, and always return an `open` func, and just have that `open` func error (as it does now). Benefit here is that in most all cases, the contracts query will succeed before a user tries to call open. So we don't impose a ux on devs to handle that case explicitly. And instead, just error when it happens. Thoughts?

Reply:
Yes the idea here is that consumers need to explicitly check for `isReady` so TS can statically check that `open` can be called.
As you described, we can alternatively always return the `open` function and in most cases it should work, but consumers that don't check for `isReady` will for sure have an error at runtime when `isReady` is `false`, and TS checks wouldn't help catch these.
I don't mind strongly either way, as you wrote the main thing here is having good docs on how to use the hook.

Reply:
Updated